File: ticket-service/src/main/java/com/yusssss/sso/ticketservice/dtos/ticket/TicketDto.java
package com.yusssss.sso.ticketservice.dtos.ticket;

import com.yusssss.sso.ticketservice.entities.Ticket;
import com.yusssss.sso.ticketservice.entities.TicketStatus;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
public class TicketDto {

    //will implement includeUser and includeEvent later
    private UUID id;

    private UUID userId;

    private UUID eventId;

    private TicketStatus status;

    private LocalDateTime createdAt;

    public TicketDto(Ticket ticket) {
        this.id = ticket.getId();
        this.userId = ticket.getUserId();
        this.eventId = ticket.getEventId();
        this.status = ticket.getStatus();
        this.createdAt = ticket.getCreatedAt();
    }
}

File: event-service/dto/ticket_dto.go
package dto

import (
	"time"

	"github.com/google/uuid"
)

type TicketStatus string

const (
	Created TicketStatus = "CREATED"
	Used    TicketStatus = "USED"
)

type TicketDto struct {
	ID        uuid.UUID    `json:"id"`
	UserId    uuid.UUID    `json:"userId"`
	EventId   uuid.UUID    `json:"eventId"`
	Status    TicketStatus `json:"status"`
	CreatedAt time.Time    `json:"createdAt"`
}
